sql/core/src/main/scala/org/apache/spark/sql/execution/ui/SQLAppStatusListener.scala

```diff
@@ -202,6 +227,47 @@ class SQLAppStatusListener(
     }
   }
 
+  /* Connects Operators to Stages by doing the following:
+   * 1. Read SparkGraph to get every Node's name and respective AccumulatorIDs.
+   * 2. Gets each stage's AccumulatorIDs.
+   * 3. Maps Operators to stages by checking for non-zero intersection of 1 and 2's AccumulatorIDs.
+   * 4. Connect SparkGraphNodes to respective StageIDs for rendering in SQL UI.
+   */
+  private def connectOperatorToStage(exec: LiveExecutionData): Map[Long, List[Int]] = {
+    // Reads in SparkPlanGraph to get each Node's name and respective accumulatorIDs
+    // and saves it inside nodeNameToAccumulatorIds.
+    val planGraph = kvstore.read(classOf[SparkPlanGraphWrapper], exec.executionId)
+      .toSparkPlanGraph()
+    val nodeNameToAccumulatorIds = planGraph.allNodes.map { node =>
+      (node.name, node.metrics.map(_.accumulatorId)) }
+    logDebug("each Operator's Metrics represented by AccumulatorIds are: \n %s\n"
+      .format(nodeNameToAccumulatorIds.mkString("\n ")))
+
+    // Gets each stage and its list of distinct accumulatorIDs
+    // which is retrived upon completion of a stage.
+    val stageIdToAccumulatorIDs = stageAccumulators.asScala
+    logDebug("each Stage's Metrics represented by AccumulatorIds are: \n %s\n"
+      .format(stageIdToAccumulatorIDs.mkString("\n ")))
+
+    // Maps stages to operators by checking for non-zero intersection
+    // between nodeMetrics and stageAccumulateIDs
+    val operatorToStage = nodeNameToAccumulatorIds.map { case (nodeName, accumulatorIds1) =>
```

Comment:
was curious how long this takes was wondering if it might be more efficient to have accumid to stage id to look it up, but need to think about it some more.

nit below the if statement should be formatted nicer in the very least.

```
if (accumulatorIds1.intersect(accumulatorIds2).nonEmpty) {
            Some(stageId)
          } else {
            None
          }
```

Reply:
reformatted the if statement. I am not sure whether having accumid to stage id to look it up would be more efficient or not.

Reply:
ok, I will see if I can run a couple tests to see how long it takes